packages/cactus-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-ledger-connector-besu/deploy-contract/get-record-locator.test.ts

```diff
@@ -0,0 +1,378 @@
+import { v4 as uuidv4 } from "uuid";
+import { PluginRegistry } from "@hyperledger/cactus-core";
+import {
+  EthContractInvocationType,
+  Web3SigningCredentialType,
+  PluginLedgerConnectorBesu,
+  PluginFactoryLedgerConnector,
+  ReceiptType,
+  InvokeContractV1Request,
+  BesuApiClientOptions,
+  BesuApiClient,
+  IPluginLedgerConnectorBesuOptions,
+  GetBesuRecordV1Request,
+} from "../../../../../main/typescript/public-api";
+import { PluginKeychainMemory } from "@hyperledger/cactus-plugin-keychain-memory";
+import {
+  BesuTestLedger,
+  pruneDockerAllIfGithubAction,
+} from "@hyperledger/cactus-test-tooling";
+import { LogLevelDesc } from "@hyperledger/cactus-common";
+import HelloWorldContractJson from "../../../../solidity/hello-world-contract/HelloWorld.json";
+import Web3 from "web3";
+import { PluginImportType } from "@hyperledger/cactus-core-api";
+import { createServer } from "http";
+import { AddressInfo } from "net";
+import {
+  ApiServer,
+  AuthorizationProtocol,
+  ConfigService,
+} from "@hyperledger/cactus-cmd-api-server";
+import { Secp256k1Keys, KeyFormat } from "@hyperledger/cactus-common";
+import KeyEncoder from "key-encoder";
+
+const testCase = "get record locator";
+const logLevel: LogLevelDesc = "TRACE";
+
+test("BEFORE " + testCase, async (t: Test) => {
+  const pruning = pruneDockerAllIfGithubAction({ logLevel });
+  await t.doesNotReject(pruning, "Pruning didn't throw OK");
+  t.end();
+});
+
+test(testCase, async (t: Test) => {
+  const httpServer = createServer();
+  await new Promise((resolve, reject) => {
+    httpServer.once("error", reject);
+    httpServer.once("listening", resolve);
+    httpServer.listen(0, "127.0.0.1");
+  });
+  const addressInfo = httpServer.address() as AddressInfo;
+  t.comment(`HttpServer AddressInfo: ${JSON.stringify(addressInfo)}`);
+  const nodeHost = `http://${addressInfo.address}:${addressInfo.port}`;
+  t.comment(`Cactus Node 1 Host: ${nodeHost}`);
+  const configuration = new BesuApiClientOptions({ basePath: nodeHost });
+  const api = new BesuApiClient(configuration);
+
+  const besuTestLedger = new BesuTestLedger();
+  await besuTestLedger.start();
+
+  test.onFinish(async () => {
+    await besuTestLedger.stop();
+    await besuTestLedger.destroy();
+  });
+
+  const rpcApiHttpHost = await besuTestLedger.getRpcApiHttpHost();
+  const rpcApiWsHost = await besuTestLedger.getRpcApiWsHost();
+
+  /**
+   * Constant defining the standard 'dev' Besu genesis.json contents.
+   *
+   * @see https://github.com/hyperledger/besu/blob/1.5.1/config/src/main/resources/dev.json
+   */
+  const firstHighNetWorthAccount = besuTestLedger.getGenesisAccountPubKey();
+  const besuKeyPair = {
+    privateKey: besuTestLedger.getGenesisAccountPrivKey(),
+  };
+  ////////////////////////////////////////////////////////////////////
+  const keyEncoder: KeyEncoder = new KeyEncoder("secp256k1");
+  const keychainId = uuidv4();
+  const keychainRef = uuidv4();
```

Comment:
@AzaharaC This keychain ref does not appear to be used (if I read the code right) and could be removed, along with the dependency in package.json itself.